nav2_bt_navigator/src/ros_console_logger.cpp

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) 2019 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "nav2_bt_navigator/ros_console_logger.hpp"
+
+namespace nav2_bt_navigator
+{
+
+RosConsoleLogger::RosConsoleLogger(const rclcpp::Logger & logger, const BT::Tree & tree)
+: StatusChangeLogger(tree.root_node), logger_(logger)
+{
+}
+
+void RosConsoleLogger::callback(
```

Comment:
I would argue you should remove `RosConsoleLogger` and just log to debug in the topic publisher. It seems too much abstraction to separate them for just printing to logs

Reply:
The console logger is much easier to read than the output of ros2 topic echo. I don't see myself wanting to use that knowing the other exists. I'll take a look at the diagnostics package and see if that is a reasonable alternative.

Reply:
Oh I agree, I just meant in the topic logger callback, you include this debug message. I think that’s totally reasonable.

Reply:
Oh. I see. That makes sense since we have both enabled at the same time anyhow. I guess I got too caught up following the upstream examples. :-)

Reply:
So is the plan here to get rid of the console logger and merge the debug messages into the topic logger? Just wondering if I should be reviewing as is or waiting for that change?

Reply:
That's my understanding as well